File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetGamesUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import java.util.List;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.GamesRepository;
import es.etologic.mahjongscoring2.domain.model.Game;
import es.etologic.mahjongscoring2.domain.model.GameWithRounds;
import io.reactivex.Completable;
import io.reactivex.Single;

public class GetGamesUseCase {

    private GamesRepository gamesRepository;

    @Inject
    public GetGamesUseCase(GamesRepository gamesRepository) {
        this.gamesRepository = gamesRepository;
    }

    public Single<List<GameWithRounds>> getAllWithRounds() {
        return gamesRepository.getAllWithRounds();
    }

    public Single<GameWithRounds> getGame(long gameId) {
        return gamesRepository.getOneWithRounds(gameId);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/GetGamesUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.GamesRepository
import es.etologic.mahjongscoring2.domain.model.GameWithRounds
import io.reactivex.Single
import javax.inject.Inject

class GetGamesUseCase @Inject
constructor(private val gamesRepository: GamesRepository) {
    
    fun getAllWithRounds(): Single<List<GameWithRounds>> = gamesRepository.getAllWithRounds()
    
    fun getGame(gameId: Long): Single<GameWithRounds> = gamesRepository.getOneWithRounds(gameId)
}
